test/e2e/adminapi_kubernetesobjects.go

```diff
@@ -0,0 +1,350 @@
+			params := url.Values{
+				"kind":      []string{"secret"},
+				"namespace": []string{namespace},
+				"name":      []string{objName},
+			}
+			var cloudErr api.CloudError
+			resp, err := adminRequest(ctx, resourceID, http.MethodGet, "/kubernetesobjects", params, nil, &cloudErr)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resp.StatusCode).To(Equal(http.StatusForbidden))
+
+			By("checking response for error")
+			Expect(cloudErr.Code).To(Equal(api.CloudErrorCodeForbidden))
+		})
+
+		It("should not be able to get a list of secrets", func() {
+			By("requesting a list of Secret objects via RP admin API")
+			params := url.Values{
+				"kind":      []string{"secret"},
+				"namespace": []string{namespace},
+			}
+			var cloudErr api.CloudError
+			resp, err := adminRequest(ctx, resourceID, http.MethodGet, "/kubernetesobjects", params, nil, &cloudErr)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resp.StatusCode).To(Equal(http.StatusForbidden))
+
+			By("checking response for error")
+			Expect(cloudErr.Code).To(Equal(api.CloudErrorCodeForbidden))
+		})
+
+		It("should not be able to delete a secret", func() {
+			By("deleting the secret via RP admin API")
+			params := url.Values{
+				"kind":      []string{"secret"},
+				"namespace": []string{namespace},
+				"name":      []string{objName},
+			}
+			var cloudErr api.CloudError
+			resp, err := adminRequest(ctx, resourceID, http.MethodDelete, "/kubernetesobjects", params, nil, &cloudErr)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resp.StatusCode).To(Equal(http.StatusForbidden))
+			Expect(cloudErr.Code).To(Equal(api.CloudErrorCodeForbidden))
+		})
+	}
+
+	testConfigMapCreateOK := func() {
+		By("creating a new object via RP admin API")
+		body := fmt.Sprintf(
+			`{
+				"kind": "ConfigMap",
+				"apiVersion": "v1",
+				"metadata": {
+					"name": "%s",
+					"namespace": "%s"
+				},
+				"data": {
+					"key": "value"
+				}
+			}`,
+			objName,
+			namespace,
+		)
+		resp, err := adminRequest(ctx, resourceID, http.MethodPost, "/kubernetesobjects", nil, json.RawMessage(body), nil)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+	}
+
+	testConfigMapGetOK := func() {
+		By("getting an object via RP admin API")
+		params := url.Values{
+			"kind":      []string{"configmap"},
+			"namespace": []string{namespace},
+			"name":      []string{objName},
+		}
+		var obj unstructured.Unstructured
+		resp, err := adminRequest(ctx, resourceID, http.MethodGet, "/kubernetesobjects", params, nil, &obj)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+		By("comparing it to the actual object retrived via Kubernetes API")
+		cm, err := clients.Kubernetes.CoreV1().ConfigMaps(namespace).Get(objName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(obj.GetNamespace()).To(Equal(cm.Namespace))
+		Expect(obj.GetName()).To(Equal(cm.Name))
+		Expect(obj.GetAnnotations()).To(Equal(cm.Annotations))
```

Comment:
would comparing the data be better?  We don't appear to set annotations on the configmap above.

Reply:
Leftover from the original PR. Will replace with data.